File: net/Azure.Storage.Blobs.PerfStress/GetBlobsTest.cs
using Azure.Storage.Blobs.PerfStress.Core;
using System;
using System.IO;
using System.Threading;
using System.Threading.Tasks;

namespace Azure.Storage.Blobs.PerfStress
{
    public class GetBlobsTest : ContainerTest<CountOptions>
    {
        public GetBlobsTest(CountOptions options) : base(options)
        {
        }

        public override async Task GlobalSetupAsync()
        {
            await base.GlobalSetupAsync();

            var uploadTasks = new Task[Options.Count];
            for (var i = 0; i < uploadTasks.Length; i++)
            {
                var blobName = "getblobstest-" + Guid.NewGuid();
                uploadTasks[i] = BlobContainerClient.UploadBlobAsync(blobName, Stream.Null);
            }
            await Task.WhenAll(uploadTasks);
        }

        public override void Run(CancellationToken cancellationToken)
        {
            // Enumerate collection to ensure all BlobItems are downloaded
            foreach (var _ in BlobContainerClient.GetBlobs(cancellationToken: cancellationToken));
        }

        public override async Task RunAsync(CancellationToken cancellationToken)
        {
            // Enumerate collection to ensure all BlobItems are downloaded
            await foreach (var _ in BlobContainerClient.GetBlobsAsync(cancellationToken: cancellationToken)) { }
        }
    }
}

File: sdk/storage/perf-test/src/main/java/com/azure/storage/blob/perfstress/GetBlobsTest.java
package com.azure.storage.blob.perfstress;

import java.util.UUID;

import com.azure.perfstress.CountOptions;
import com.azure.storage.blob.models.ParallelTransferOptions;
import com.azure.storage.blob.perfstress.core.ContainerTest;

import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public class GetBlobsTest extends ContainerTest<CountOptions> {
    public GetBlobsTest(CountOptions options) {
        super(options);
    }

    public Mono<Void> GlobalSetupAsync() {
        return super.GlobalSetupAsync().then(
            Flux.range(0, Options.Count)
                .map(i -> "getblobstest-" + UUID.randomUUID())
                .flatMap(b -> BlobContainerAsyncClient.getBlobAsyncClient(b).upload(Flux.empty(), null))
                .then());
    }

    @Override
    public void Run() {
        BlobContainerClient.listBlobs().forEach(b -> {});
    }

    @Override
    public Mono<Void> RunAsync() {
        return BlobContainerAsyncClient.listBlobs().then();
    }
}